scripts/run-sonata-sim.sh

```diff
@@ -25,27 +25,20 @@ fi
 # Remove old uart log
 rm -f "${SONATA_SIMULATOR_UART_LOG}"
 
-# If a second argument is provided, check content of UART log.
-if [ -n "$2" ] ; then
-	# Run the simulator in the background.
-	${SONATA_SIMULATOR} -E "${SONATA_SIMULATOR_BOOT_STUB}" -E "$1" &
-	LOOP_TRACKER=0
-	while (( LOOP_TRACKER <= 60 ))
-	do
-		sleep 1s
-		# Returns 0 if found and 1 if not.
-		MATCH_FOUND=$(grep -q -F -f "$2" "${SONATA_SIMULATOR_UART_LOG}"; echo $?)
-		if (( MATCH_FOUND == 0 )) ; then
-			# Match was found so exit with success
-			pkill -P $$
-			exit 0
-		fi
-		LOOP_TRACKER=$((LOOP_TRACKER+1))
-	done
-	# Timeout was hit so no success.
-	pkill -P $$
+# Run the simulator with a timeout and capture the exit status. The || is needed so that set -e doesn't kill us if this fails
+# and we can report the error gracefully. The timeout is chosen so that the testsuite can run in CI. It's probably
+# be a bit long for other purposes.
+result=0
+timeout --foreground 20m ${SONATA_SIMULATOR} -E "${SONATA_SIMULATOR_BOOT_STUB}" -E "$1" || result=$?
```

Comment:
Please don't depend on non-POSIX things.  If we need to use timeout in CI, that's fine (it's a known environment) but requiring developers to install GPLv3 things is mean.